loki/analyse/util_linear_algebra.py

```diff
@@ -0,0 +1,193 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from numpy import zeros_like, dot
+from numpy import vstack, hstack
+from numpy import all as np_all, sum as np_sum, isin as np_isin
+
+__all__ = [
+    "back_substitution",
+    "generate_row_echelon_form",
+    "is_independent_system",
+    "yield_one_d_systems",
+]
+
+
+def is_independent_system(matrix):
+    """
+    Check if a linear system of equations can be split into independent one-dimensional problems.
+
+    Args:
```

Comment:
Would you mind changing the docstring style to the numpy layout in all routines?
https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html

Reply:
7de1b4390cc64c2e1895f72356de18b883b4a6f7